### DISCUSSION TITLE
Remove item from array based on it's value.

### DISCUSSION THREAD

**Discussion by nitanmarcel** [nlohmann/json#3488] [Q&A] [answered]
Having the json bellow, how can I get the array containing `My Game 2` then delete the entire array from my json object?

The json array can get bigger so the item is not always in the same position so before deleting it I have to find it's index.

```
[
    {
        "title": "My Game",
        "url": "file://* or http*"
    },
    {
        "title": "My Game2",
        "url": "file://* or http*"
    }
]
```

**Comment by falbrechtskirchinger** [comment, marked as answer]
Try this:
- Iterate through the array.
- Find the object where the key "title" has the value "My Game2" and stop iterating.
- If the iterator is not the past-the-end iterator, erase the object at that location.

```cpp
auto it = j.begin();
for(; it != j.end(); ++it) {
    if(it->at("title") == "My Game2")
        break;
}

if(it != j.end()) j.erase(it);
```

https://json.nlohmann.me/api/basic_json/begin/
https://json.nlohmann.me/api/basic_json/at/
https://json.nlohmann.me/api/basic_json/erase/
